data/soapvalidator/Mail/Persona/SendOnBehalfOfMailUsingPrimaryAccount.xml

```diff
@@ -0,0 +1,372 @@
+				</DiscoverRightsRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//acct:DiscoverRightsResponse/acct:targets"
+					attr="right" match="sendOnBehalfOf" />
+				<t:select
+					path="//acct:DiscoverRightsResponse/acct:targets/acct:target/acct:email[@addr='${test_account2.name}']" />
+			</t:response>
+		</t:test>
+
+		<!-- Send email with From: Primary Account -->
+		<t:test id="sendmail1" required="true">
+			<t:request>
+				<SendMsgRequest xmlns="urn:zimbraMail">
+					<m>
+						<e t="f" a="${test_account2.name}" />
+						<e t="t" a="${test_account3.name}" />
+						<su> ${msg01.subject}</su>
+						<mp ct="text/plain">
+							<content> ${message.content1}</content>
+						</mp>
+					</m>
+				</SendMsgRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//zimbra:Code" match="service.PERM_DENIED"
+					emptyset="1" />
+				<t:select path="//mail:SendMsgResponse/mail:m" attr="id"
+					set="message.id1" />
+			</t:response>
+		</t:test>
+
+		<!-- Verify message details using GetMsgRequest -->
+		<t:test id="GetMsgRequest">
+			<t:request>
+				<GetMsgRequest xmlns="urn:zimbraMail">
+					<m id="${message.id1}" />
+				</GetMsgRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//mail:GetMsgResponse/mail:m[@id='${message.id1}']">
+					<t:select path="//mail:e[@t='f']" attr="a"
+						match="${test_account2.name}" />
+					<t:select path="//mail:e[@t='t']" attr="a"
+						match="${test_account3.name}" />
+					<t:select path="//mail:su" match="${msg01.subject}" />
+				</t:select>
+			</t:response>
+		</t:test>
+
+		<!-- Login as recipient test_account3.name -->
+		<t:property name="server.zimbraAccount" value="${test_acct3.server}" />
+		<t:test id="auth_acct3" required="true" depends="acct1_setup">
+			<t:request>
+				<AuthRequest xmlns="urn:zimbraAccount">
+					<account by="name">${test_account3.name}</account>
+					<password>${defaultpassword.value}</password>
+				</AuthRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//acct:AuthResponse/acct:lifetime" match="^\d+$" />
+				<t:select path="//acct:AuthResponse/acct:authToken" set="authToken" />
+			</t:response>
+		</t:test>
+
+		<t:test>
+			<t:request>
+				<SearchRequest xmlns="urn:zimbraMail" types="message">
+					<query>from:(${test_account2.name})</query>
+				</SearchRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//mail:SearchResponse/mail:m/mail:su"
+					match="${msg01.subject}" />
+				<t:select path="//mail:SearchResponse/mail:m" attr="id"
+					set="msg01.id" />
+			</t:response>
+		</t:test>
+
+		<!-- Verify message details using GetMsgRequest -->
+		<t:test id="GetMsgRequest">
+			<t:request>
+				<GetMsgRequest xmlns="urn:zimbraMail">
+					<m id="${msg01.id}" />
+				</GetMsgRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//mail:GetMsgResponse/mail:m[@id='${msg01.id}']">
+					<t:select path="//mail:e[@t='f']" attr="a"
+						match="${test_account2.name}" />
```

Comment:
Add validation for checking @t='s' element. In case of SOBO, sender value will be updated in this element.